### PR TITLE
fix: macOS ApplicationTheme observing

### LINKED ISSUE #3619: [macOS] Changing application theme at runtime causes exception
## Current behavior

When `Application.RequestedTheme` is not set, changing the theme at runtime causes an exception.

## Expected behavior

No exception.

## Environment

Nuget Package:

Package Version(s):

Affected platform(s):

- [ ] iOS
- [ ] Android
- [ ] WebAssembly
- [ ] WebAssembly renderers for Xamarin.Forms
- [x] macOS
- [ ] Windows
- [ ] Build tasks
- [ ] Solution Templates

Visual Studio:

- [ ] 2017 (version: )
- [ ] 2019 (version: )
- [x] for Mac (version: )

Relevant plugins:

- [ ] Resharper (version: )

### PR DESCRIPTION
**Pull request by MartinZikmund**
GitHub Issue (If applicable): fixes #3619

## PR Type

What kind of change does this PR introduce?

- Bugfix

## What is the current behavior?

Theme changes at runtime crash macOS app.

## What is the new behavior?

Theme observing works.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [x] Validated PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [x] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

### CODE REVIEW

File: src/Uno.UI/UI/Xaml/Application.macOS.cs [hunk 1]
```diff
@@ -154,11 +171,15 @@ private void SetCurrentLanguage()
 		}
 
 		partial void ObserveSystemThemeChanges()
-		{			
-			NSUserDefaults.StandardUserDefaults.AddObserver(
-				"AppleInterfaceStyle",
-				NSKeyValueObservingOptions.New,
-				_ => Application.Current.OnSystemThemeChanged());
+		{
+			NSDistributedNotificationCenter.GetDefaultCenter().AddObserver(
```
Comment (jeromelaban) [inline src/Uno.UI/UI/Xaml/Application.macOS.cs:175]: Was that related to a race condition ? or just based on OS version?

File: src/Uno.UI/UI/Xaml/Application.macOS.cs [hunk 2]
```diff
@@ -110,31 +114,44 @@ private bool TryHandleUrlActivation(NSUrl[] urls, ApplicationExecutionState prev
...
 			}
-			if (NSUserDefaults.StandardUserDefaults["AppleInterfaceStyle"] == null)
-			{
-				return ApplicationTheme.Light;
-			}
-			else
+			return ApplicationTheme.Light;
+		}
+
+		private Version GetSystemVersion()
```
Comment (jeromelaban) [inline src/Uno.UI/UI/Xaml/Application.macOS.cs:134]: That should probably go into another helper class in Uno, or an existing API that provides that.

File: src/Uno.UI/UI/Xaml/Application.macOS.cs (hunk 1 above)
Comment (MartinZikmund) [reply src/Uno.UI/UI/Xaml/Application.macOS.cs:175]: Apparently this is the only way that works properly 😄 - https://stackoverflow.com/a/53358384/732221. Also he mentions correctly, that all other overloads of the AddObserver method are broken in Xamarin.

File: src/Uno.UI/UI/Xaml/Application.macOS.cs (hunk 2 above)
Comment (MartinZikmund) [reply src/Uno.UI/UI/Xaml/Application.macOS.cs:134]: Will check if it is somewhere 👍

Comment (MartinZikmund) [reply src/Uno.UI/UI/Xaml/Application.macOS.cs]: Fixed

**Pull request merged**